SemiconductorTestLibrary.Extensions/source/InstrumentAbstraction/DAQmx/AnalogInput.cs

```diff
@@ -22,6 +22,7 @@ public static PinSiteData<double[]> ReadAnalogMultiSample(this DAQmxTasksBundle
             return tasksBundle.DoAndReturnPerSitePerPinResults(taskInfo =>
             {
                 taskInfo.VerifyTaskType(DAQmxTaskType.AnalogInput);
+                taskInfo.Task.Stream.ChannelsToRead = taskInfo.ChannelList;
```

Comment:
I think we need to restore original channels after the read operation, otherwise subsequent reads using direct driver calls will be affected.

Reply:
@lornazh Updated as such, I have introduced new variable to store the All channels information.

 As strings are immutable I've taken this approach. Since the type `Tasks` are of type mutable, it is taking shallow copy of the value that is taken.

Please let me know if there are any concerns.